### PR TITLE
Added makefile to build examples

### LINKED ISSUE #129: Makefile which includes children folder's makefiles
We have many examples in `examples` folder. Each folder contains one or multiple projects. We need to add a single makefile in `examples` folder which could execution the compilation of every individual makefile in all folders and do **not** stop if one of projects fails to compile. Ideally we need to output the number of compiled projects/examples versus overall projects and list failed projects.

### PR DESCRIPTION
**Pull request by hrishabh23**
Added a makefile which uses recursive make to build example programs using their own makefiles.
The makefile does not stop on failed compilations.
Fixes #129

### CODE REVIEW

**Comment by ihhub** [comment]
How about the rest of examples like `thread_pool`, `cuda` and etc.?

File: examples/makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+##
+# Directories containing makefiles
+##
+subsystem:
+	-$(MAKE) -C bitmap_operation
+	-$(MAKE) -C blob_detection
+	-$(MAKE) -C function_pool
+	-$(MAKE) -C image_function
+	-$(MAKE) -C median_filter
+
```
Comment (ihhub) [inline examples/makefile]: -$(MAKE) -C opencl/image_function
	-$(MAKE) -C thread_pool

File: examples/makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+.PHONY: clean
+clean:
+	-$(MAKE) clean -C bitmap_operation
+	-$(MAKE) clean -C blob_detection
+	-$(MAKE) clean -C function_pool
+	-$(MAKE) clean -C image_function
+	-$(MAKE) clean -C median_filter
```
Comment (ihhub) [inline examples/makefile]: -$(MAKE) clean -C opencl/image_function
	-$(MAKE) clean -C thread_pool

**Comment by ihhub** [comment]
Thanks for contribution to the project!

**Comment by hrishabh23** [comment]
Apologies for missing some programs at first.

**Pull request merged**